src/products/schemas/product.schema.ts

```diff
@@ -0,0 +1,13 @@
+import * as mongoose from 'mongoose';
+
+export const ProductSchema = new mongoose.Schema(
+  {
+    name: String,
+    price: Number,
+    stock: Number,
+    status: String,
```

Comment:
I think you can use it that way
```suggestion
    status: {
      type: String,
      enum: Object.values(StatusEnum),
    },
```